core/src/main/java/com/sekwah/advancedportals/core/portal/AdvancedPortal.java

```diff
@@ -146,6 +147,11 @@ public ActivationResult activate(PlayerContainer player,
 
         PlayerData playerData = playerDataServices.getPlayerData(player);
 
+        if (!Permissions.USE_PORTAL.hasPermission(player)) {
+            player.sendMessage(Lang.translate("portal.error.nopermission"));
+            return ActivationResult.FAILED_DO_NOTHING;
```

Comment:
Every  other portal failure with a message knocks you back, would probably be best to do this for consistency